tasks/install_system.yml

```diff
@@ -0,0 +1,68 @@
+- name: Make sure iscsi device for {{ vm.host }} @{{ vm.org }} is logged in
+  systemd:
+    name: vm_iscsi@{{ vm.org }}/{{ vm.host }}.service
+    state: started
+  when: vm.storage_type|default(xen_vman_storage_type) == "iscsi"
+
+- name: Format root storage for {{ vm.host }} @{{ vm.org }}
+  shell: mkfs.{{ vm.filesystem|default(xen_vman_default_filesystem) }} "/dev/disk/by-path/ip-{{ xen_vman_iscsi_server }}-iscsi-{{ xen_vman_iscsi_base_wwn }}:{{ vm.org }}-{{ vm.host }}-lun-0"
+  when: vm.storage_type|default(xen_vman_storage_type) == "iscsi"
+
+- name: Make sure vm root is mounted
+  systemd:
+    name: vm_mount@{{ vm.org }}/{{ vm.host }}.service
+    state: started
+
+# TODO: make system installation async
+- name: Install {{ vm.os.type }} {{ vm.os.version }} at {{ vm.host }} @{{ vm.org }}
+  shell: debootstrap --include openssh-server,python {{ vm.os.version }} /mnt/vms/{{ vm.org }}-{{ vm.host }}
+  when: vm.os.type == "ubuntu"
+
+- name: Create current ansible user at {{ vm.host }} @{{ vm.org }}
+  shell: chroot /mnt/vms/{{ vm.org }}-{{ vm.host }} /usr/sbin/useradd -c "Create for inital use by xen_vman" -G "users,sudo" -m "{{ ansible_user }}"
+  when: vm.os.type == "ubuntu"  # This check is needed since different distributions use different default groups for sudo
```

Comment:
Are we currently supporting different distributions?

Reply:
Yes, you can use this role even to manage arch or gentoo vms, but they aren't automated installed.